src/Framework/TestResult.php

```diff
@@ -229,21 +234,21 @@ public function addError(Test $test, Throwable $t, float $time): void
                 $test->markAsRisky();
             }
 
-            if ($this->stopOnRisky) {
+            if ($this->stopOnRisky || $this->stopOnDefect) {
                 $this->stop();
             }
         } elseif ($t instanceof IncompleteTest) {
             $this->notImplemented[] = new TestFailure($test, $t);
             $notifyMethod           = 'addIncompleteTest';
 
-            if ($this->stopOnIncomplete) {
+            if ($this->stopOnIncomplete || $this->stopOnDefect) {
                 $this->stop();
             }
         } elseif ($t instanceof SkippedTest) {
             $this->skipped[] = new TestFailure($test, $t);
             $notifyMethod    = 'addSkippedTest';
 
-            if ($this->stopOnSkipped) {
+            if ($this->stopOnSkipped || $this->stopOnDefect) {
```

Comment:
to be honest not sure about this...
basically, it means that I will **never** use `--stopOnDefect`
in bigger projects, there are always some tests depending on environment, eg different version of PHP, OS, ext, ini...

Reply:
Yes, thanks for noticing! This is also a huge doubt of mine. On the one hand I want to have a shorthand for "I just want a quick edit-test-edit-test cycle right now", but it has also gotten in my way so much I had moved the "not really bad" defects down the sorter list.

For now I will remove the halting part and leave it up to the user to supply explicit `--stop-on-warning`, etc options

Reply:
please dont
(elaboration soon)

Reply:
stopping on given type of result (~defect) shall work the same as putting them to cache to be executed first.

it's great value for me to have `--stop-on-defect` rather than `--stop-on-*` 10 times.

Only thing is to properly specify what is a defect.
For me, it all that "executed and failed" - thus all that need to manual fix from developer.
Thus, not met expectation, warning, ..., but not "skipped" or "incomplete"

Reply:
I like this suggestion very much: this was the first version I made. Remember that code style fix for the `defectsWeights` constant? That was a leftover of the original configurable version which I thought overcomplicated at the time.

I will look for a nice way to add it back in.

Reply:
Rereading your comment: things that need a manual fix from the developer in the test or source would be:
- error: usually PHP syntax mistakes
- failure: assert failing
- risky: developer forgot to add assertions to a new test

Am I reading you correctly?

Reply:
exactly.

to clarify - i'm not thinking about making it configurable again, it shall not.
defects are all that require developer fix, so exactly ones you listed :+1: